packages/nodeconn/chain.go

```diff
@@ -93,6 +93,24 @@ func (ncc *ncChain) postTxLoop(ctx context.Context) {
 		if err != nil {
 			return nil, err
 		}
+
+		dryRes, err := ncc.nodeConn.httpClient.DryRunTransaction(task.ctx, txBytes)
```

Comment:
Why is dry-run needed? The result is only used for logging? Maybe explain in a comment :pray:

Reply:
DryRun runs the transaction without persisting it. 

If we ever face an issue where our Anchor progress transaction is faulty - like a MoveCall incorrectly set - it will get executed, fail and it will cost Gas. The nodes will then retry and it will do so until the Gascoin is drained.

DryRun before makes sure that the TX is correct, as we can check the Effects for success. And only afterwords we send the TX away.

Reply:
:+1: 

```suggestion
                // Dry-run first to avoid spending gas if the tx fails
		dryRes, err := ncc.nodeConn.httpClient.DryRunTransaction(task.ctx, txBytes)
```